pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -383,6 +384,17 @@ final class Event {
           'surveyId': surveyId,
         };
 
+  @override
+  int get hashCode => eventData.hashCode;
+
+  @override
+  bool operator ==(Object other) =>
+      other is Event &&
+      other.runtimeType == runtimeType &&
+      other.eventName == eventName &&
+      other.eventData == eventData &&
+      compareEventData(other.eventData, eventData);
+
```

Comment:
I've added these overrides for now and will add some tests to ensure it works correctly, but does this look good? @bkonyi 

We essentially just need to check the event name and the data in the `eventData` map